src/pipecat/services/whisper_base.py

```diff
@@ -0,0 +1,94 @@
+#
+# Copyright (c) 2025, Daily
+#
+# SPDX-License-Identifier: BSD 2-Clause License
+#
+
+from typing import AsyncGenerator, Optional
+
+from loguru import logger
+
+from pipecat.frames.frames import ErrorFrame, Frame, TranscriptionFrame
+from pipecat.services.ai_services import SegmentedSTTService
+from pipecat.utils.time import time_now_iso8601
+
+try:
+    from openai import AsyncOpenAI
+    from openai.types.audio import Transcription
+except ModuleNotFoundError as e:
+    logger.error(f"Exception: {e}")
+    logger.error(
+        "In order to use OpenAI, you need to `pip install pipecat-ai[openai]`. Also, set `OPENAI_API_KEY` environment variable."
+    )
+    raise Exception(f"Missing module: {e}")
+
+
+class BaseWhisperSTTService(SegmentedSTTService):
```

Comment:
maybe rename file to `base_whisper.py`? we usually do that in all the other base classes.